Multi-product release docs (#728)
mattermost/mattermost-handbook: operations/research-and-development/product/release-process/incident-collaboration-release.md

## Release Timeline

Schedule for Incident Collaboration releases:
 - Features and bug fixes are merged
 - (T-3): Cut a Release Candidate
 - (T-2): Smoketesting
 - (T-0): Release Day

On the first Monday of each month, or the next business day, the Incident Collaboration team's Product Manager and Dev Lead choose what version of the plugin they will pre-package into that month's self-managed release. The deadline to merge the selected version into the self-managed release is the Code Freeze day. An additional round of testing is required to qualify before merging.

## Notes

 - Critical bugs are flagged and fixed before releasing, with quick dot releases completed as needed.
 - The team uses Incident Collaboration Playbook checklists for releases and for pre-packaging with self-managed.

## Basic release tasks

 - Track and triage bugs via Jira
 - Add PR milestones for QA tracking
 - Auto-generated release notes with changes, upgrade notes, minimum server version requirement, contributors, and known issues
 - Update NOTICE.txt
 - Update dependencies

>  - Testing + automated tests

Reviewer: ```suggestion
 - Testing and automated tests
```

 - Marketing - Release announcements, tweets
 - Feature documentation
 - Release

>     - Github release

Reviewer: ```suggestion
    - GitHub release
```

    - Check minimum required server version (if applicable)
 - Contributors
    - Swag
 - Release dates communicated in Incident Response google calendar
 - Release retrospective

>  - **Security updates communication XXXX**

Author: @lieut-data Are there normally any security fixes for IC?

Reviewer 2: We haven't yet had to flesh out a process here, no, but I anticipate we would look to the integrations team experience on this front.

> # Incident Collaboration Release Process

Author: @lieut-data I created a basic outline of the IC release process below, please review for any suggested edits.

This document outlines the Incident Collaboration release process for the Mattermost core team. Our release process draws from many best practice Agile, Scrum, and Software Development Lifecycle methodologies.

> Every two weeks, the team runs through a playbook to cut a new release of the plugin, smoke test it, and add it to the marketplace. That gets pushed to cloud immediately (via split.io), and any self-managed customers who desire the upgrade may do so from the plugin marketplace. Weekly releases may be a future consideration, but not a plan yet.

Reviewer 2: Propose we drop mention of weekly releases for now:

```suggestion
Every two weeks, the team runs through a playbook to cut a new release of the plugin, smoke test it, and add it to the marketplace. That gets pushed to cloud immediately (via split.io), and any self-managed customers who desire the upgrade may do so from the plugin marketplace.
```

## Release Timeline

Schedule for Incident Collaboration releases:
 - Features and bug fixes are merged
 - (T-3): Cut a Release Candidate
 - (T-2): Smoketesting
 - (T-0): Release Day

> On the first day of each month, the Incident Collaboration team's Product Manager and Dev Lead choose which recent release they will include in the next self-managed release pre-packaged version. The deadline to get the pre-packaged version to the self-managed release is by self-managed Code Freeze day. The pre-packaged version is tested a few days prior to self-managed Code Freeze day.

Reviewer 2: ```suggestion
On the first Monday of each month, or the next business day, the Incident Collaboration team's Product Manager and Dev Lead choose what version of the plugin they will pre-package into that month's self-managed release. The deadline to merge the selected version into the self-managed release is the Code Freeze day. An additional round of testing is required to qualify before merging.
```

## Notes

 - Critical bugs are flagged and fixed before releasing, with quick dot releases completed as needed.
 - The team uses Incident Collaboration Playbook checklists for releases and for pre-packaging with self-managed.

>  - Incident Collaboration is deployed as a plugin, so the team is free to ship as often as they want, and without even being tied to the cloud release cadence.
>  - TBD: The team is entertaining the idea of deploying to -daily on a near daily basis, though haven't invested in the infrastructure there yet.

Reviewer 2: Propose just removing this for now:
```suggestion
```

## Basic release tasks

 - Track and triage bugs via Jira
 - Add PR milestones for QA tracking

>  - Update changelog with changes, upgrade notes, contributors, and known issues

Reviewer 2: Right now, we mostly rely on the auto-generated release notes that come from PR commits. Should we enhance this process somehow?

Author: I added a note that it's auto-generated. Would adding contributors and known issues be a future consideration?

>  - Update NOTICE.txt

Reviewer 2: I still need to get back to this! Thanks for the reminder :)

 - Update dependencies

>  - List software requirements (e.g. OS, browser) and minimum server version requirements

Reviewer 2: Wondering how to communicate this? In general, we should match with the server versions that we support.

Author: The min server version listed in the changelog is fine.
